tests/func/metrics/test_show.py

```diff
@@ -173,24 +192,33 @@ def test_show_no_repo(tmp_dir):
 
     dvc = Repo(uninitialized=True)
 
-    dvc.metrics.show(targets=["metrics.json"])
+    assert dvc.metrics.show(targets=["metrics.json"]) == {
+        "": {"data": {"metrics.json": {"data": {"foo": 0, "bar": 0.0}}}}
+    }
 
 
 def test_show_malformed_metric(tmp_dir, scm, dvc, caplog):
     tmp_dir.gen("metric.json", '{"m":1')
 
-    with pytest.raises(NoMetricsParsedError):
-        dvc.metrics.show(targets=["metric.json"])
+    assert isinstance(
+        dvc.metrics.show(targets=["metric.json"])[""]["data"]["metric.json"][
+            "error"
+        ],
+        YAMLFileCorruptedError,
+    )
+
 
+def test_metrics_show_no_target(tmp_dir, dvc, caplog):
+    with caplog.at_level(logging.WARNING):
+        assert dvc.metrics.show(targets=["metrics.json"]) == {"": {}}
 
-def test_metrics_show_no_target(tmp_dir, dvc):
-    with pytest.raises(MetricDoesNotExistError):
-        dvc.metrics.show(targets=["metrics.json"])
+    assert (
+        "'metrics.json' was not found in current workspace." in caplog.messages
+    )
 
 
 def test_show_no_metrics_files(tmp_dir, dvc, caplog):
```

Comment:
What is the behavior now if there are no metrics?

Reply:
@dberenbaum nothing, we will get no outuput. I am still working to bring back warning in case of exceptions and no results, as I did in [previous approach](https://github.com/iterative/dvc/pull/5984#issuecomment-861433436).

Reply:
warnings should be working as in the comment

Reply:
Can/should we assert that the warning is raised?

Reply:
@dberenbaum I guess this is a matter of discussion, whether we should assert such things as warnings. But I did it anyway, mostly because if we don't do that we have to check it manually.

Tests verifying the behavior are named `test_log_errors` and are created for all of `metrics/params/plots`. I did not create this mechansim for experiments, since the errors will be indicated in the table anyway.